### PR TITLE
Fix: remove permissions for Authenticated users

### PR DESCRIPTION
**Pull request by lchico**
|Related issue|
|---|
|https://github.com/wazuh/internal-devel-requests/issues/1091|

## Description
The PR removes all permissions from the ossec-agent's folder and sets Administrator(Full control), System(Full control), and for the user(Read&Execute) permissions. 

## Tests

|                                                  C:\Program Files (x86)\ossec-agent                                                  |                                                               C:\ossec-agent                                                               | 
|:------------------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------:|
| ossec_default_path | ossec_on_root_path |  
|                                                  **C:\Users\Public\ossec-agent**                                                  |                                                               **C:\Users\vagrant\Desktop\ossec-agent**                  | 
| ossec_public_path | ossec_on_Desktop_path |  

## Test Upgrade

|    Verssion 4.6.0               |            Verssion 4.9.0      |
|:------------------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------------------------------------------------------:|
C:\Program Files (x86)\ossec-agent    |  C:\Program Files (x86)\ossec-agent      | 
| ossec_default_path | ossec_on_root_path |  
 Check subfolders:                                            |C:\Program Files (x86)\ossec-agent\active-response |
| ossec_default_path | ossec_on_root_path |  
| Different Path:   |  C:\ossec-agent      | 
| ossec_root_path_4_6 | ossec_on_root_path_4_9 |  
 Check subfolders:                                            |C:\ossec-agent\active-response |
| ossec_root_path_4_6 | ossec_on_root_path_4_9 |  

- Compilation without warnings in every supported platform
  - [x] Windows
- [x] Package installation
- [x] Package upgrade

### CODE REVIEW

**Review by MarcelKemp — DISMISSED** [review]
LGTM!

**Comment by MarcelKemp** [comment]
## Failing checks
- **_Integration tests for `<module>` on Windows - Tier 0 and 1_**

It seems that all ITs are failing due to reduced permissions, because the log monitor does not have access to the `ossec.log` file.

**Comment by lchico** [comment]
> [](/apps/github-actions) ** Integration tests for enrollment on Windows - Tier 0 and 1 / run-test (pull_request) **



> access to the `ossec.log`
# Update 
By adding the (OI)(CI) flags, we can ensure that both new files and new subfolders inherit these permissions. This fix the access issue with the ossec.log file. Should we implement this change or implement a new change on the test side?

File: src/win32/InstallerScripts.vbs [hunk 1]
```diff
@@ -345,11 +345,14 @@ public function config()
...
-        setPermsInherit = "icacls """ & install_dir & """ /inheritancelevel:d /q"
+        setPermsInherit = "icacls """ & install_dir & """ /inheritancelevel:r /q"
         WshShell.run setPermsInherit, 0, True
 
-        remUserPerm = "icacls """ & install_dir & """ /remove *S-1-5-32-545 /q"
-        WshShell.run remUserPerm, 0, True
+        grantAdminPerm = "icacls """ & install_dir & """ /grant *S-1-5-32-544:(OI)(CI)F /t"
+        WshShell.run grantAdminPerm, 0, True
+
+        grantSystemPerm = "icacls """ & install_dir & """ /grant *S-1-5-18:(OI)(CI)F /t"
```
Comment (MarcelKemp) [inline src/win32/InstallerScripts.vbs:354]: Ditto:
```suggestion
        grantSystemPerm = "icacls """ & install_dir & """ /grant *S-1-5-18:(OI)(CI)F"
```

**Review by MarcelKemp — CHANGES_REQUESTED** [review]

**Review by MarcelKemp — APPROVED** [review]

**Pull request merged**